src/hct_mis_api/apps/payment/api/views.py

```diff
@@ -163,3 +169,45 @@ def _get_action_permission(self, action_name: str) -> Optional[str]:
             PaymentPlan.Action.REVIEW.name: Permissions.PM_ACCEPTANCE_PROCESS_FINANCIAL_REVIEW.name,
         }
         return action_to_permissions_map.get(action_name)
+
+
+class PaymentPlanSupportingDocumentUploadView(APIView):
+    permission_classes = [PaymentPlanSupportingDocumentUploadPermission]
+
+    def post(self, request: Request, business_area: str, program_id: str, payment_plan_id: str) -> Response:
+        payment_decode_id = decode_id_string(payment_plan_id)
+        payment_plan = get_object_or_404(PaymentPlan, id=payment_decode_id)
+
+        serializer = PaymentPlanSupportingDocumentSerializer(data=request.data, context={"payment_plan": payment_plan})
+        if serializer.is_valid():
+            serializer.save(payment_plan=payment_plan)
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+class PaymentPlanSupportingDocumentView(APIView):
+    permission_classes = []
```

Comment:
You could use `ActionMixin` and map the action with permissions here instead of separately inside function